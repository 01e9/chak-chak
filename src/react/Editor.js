import React from 'react'
import PropTypes from 'prop-types'
import { withStyles } from '@material-ui/core'

class Editor_ extends React.Component {
    static propTypes = {
        classes: PropTypes.object.isRequired,
        Canvas: PropTypes.func.isRequired,
        Toolbar: PropTypes.func.isRequired,
        ActiveTool: PropTypes.func.isRequired
    }

    render() {
        const { Canvas, Toolbar, ActiveTool, classes } = this.props;

        return (
            <React.Fragment>
                <div className={classes.canvasWrap}>
                    <Canvas/>
                    <ActiveTool/>
                </div>
                <Toolbar/>
            </React.Fragment>
        )
    }
}

const styles = {
    canvasWrap: {
        position: 'relative'
    }
}

const Editor = withStyles(styles)(Editor_);

export default Editor;
