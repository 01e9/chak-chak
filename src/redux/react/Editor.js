import Editor_ from "@/react/Editor";
import { connect } from 'react-redux'
import MainCanvas from "@/redux/react/MainCanvas";
import Toolbar from "@/redux/react/Toolbar";

const mapStateToProps = (state, ownProps) => ({
    Canvas: MainCanvas,
    Toolbar: Toolbar,
    ActiveTool: () => '[ActiveTool]' // fixme
})

const Editor = connect(mapStateToProps, null)(Editor_)

export default Editor
