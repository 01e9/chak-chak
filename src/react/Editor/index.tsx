import * as React from 'react'
import * as styles from "./styles.scss";

export interface IEditorProps {
    Canvas: React.JSXElementConstructor<{}>,
    Toolbar: React.JSXElementConstructor<{}>,
    ActiveTool: React.JSXElementConstructor<{}>
}

class Editor extends React.Component<IEditorProps> {
    render() {
        const { Canvas, Toolbar, ActiveTool } = this.props;

        return (
            <React.Fragment>
                <div className={styles.canvasWrap}>
                    <Canvas/>
                    <ActiveTool/>
                </div>
                <Toolbar/>
            </React.Fragment>
        )
    }
}

export default Editor;
