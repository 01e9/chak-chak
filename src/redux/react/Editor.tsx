import * as React from "react";
import Editor, { IEditorProps } from "@/react/Editor";
import { connect } from 'react-redux'
import Canvas from "@/redux/react/MainCanvas";
import Toolbar from "@/redux/react/Toolbar";
import ActiveTool from "@/redux/react/ActiveTool";

const mapStateToProps = (): IEditorProps => ({
    Canvas,
    Toolbar,
    ActiveTool
})

export default connect(mapStateToProps)(Editor)
