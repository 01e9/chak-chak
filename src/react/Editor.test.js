import React from 'react'
import Editor from '@/react/Editor'
import { createRender } from '@material-ui/core/test-utils';

const requiredProps = {
    Canvas: () => '[Canvas]',
    Toolbar: () => '[Toolbar]',
    ActiveTool: () => '[ActiveTool]',
}

function createRendered(props = {}) {
    return createRender()(<div><Editor {...requiredProps} {...props} /></div>)
}

describe('snapshot', () => {
    it('required props', () => {
        expect(createRendered().html()).toMatchSnapshot();
    })
})