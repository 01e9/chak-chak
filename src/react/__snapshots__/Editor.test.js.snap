// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot required props 1`] = `

<div class="Editor_-canvasWrap-1">
  [Canvas][ActiveTool]
</div>
[Toolbar]

`;
